cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
I'm in favor of reverting the changes to this file. It was meant to be a [temporary](https://github.com/kubernetes/kubernetes/pull/24124#issuecomment-214926660) fix and it's not worth plumbing logic through multiple other cloud providers to accommodate. #25029